sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,176 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from typing import TYPE_CHECKING
+from typing import cast, Optional, Union, TYPE_CHECKING
+
+from cryptography.hazmat.primitives.asymmetric.padding import AsymmetricPadding, OAEP, PKCS1v15, PSS, MGF1
+from cryptography.hazmat.primitives.asymmetric.rsa import (
+    RSAPrivateKey,
+    RSAPrivateNumbers,
+    RSAPublicKey,
+    RSAPublicNumbers,
+)
+from cryptography.hazmat.primitives.asymmetric.utils import Prehashed
+from cryptography.hazmat.primitives.hashes import Hash, HashAlgorithm, SHA1, SHA256, SHA384, SHA512
+from cryptography.hazmat.primitives.serialization import Encoding, KeySerializationEncryption, PrivateFormat
+
+from ._enums import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
 
 if TYPE_CHECKING:
-    from . import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
-    from typing import Any, Optional
+    # Import clients only during TYPE_CHECKING to avoid circular dependency
+    from ._client import CryptographyClient
+    from .._client import KeyClient
+
+
+SIGN_ALGORITHM_MAP = {
+    SHA256: SignatureAlgorithm.rs256,
+    SHA384: SignatureAlgorithm.rs384,
+    SHA512: SignatureAlgorithm.rs512,
+}
+OAEP_MAP = {
+    SHA1: EncryptionAlgorithm.rsa_oaep,
+    SHA256: EncryptionAlgorithm.rsa_oaep_256
+}
+PSS_MAP = {
+    SignatureAlgorithm.rs256: SignatureAlgorithm.ps256,
+    SignatureAlgorithm.rs384: SignatureAlgorithm.ps384,
+    SignatureAlgorithm.rs512: SignatureAlgorithm.ps512,
+}
+
+
+class ManagedRsaKey(RSAPrivateKey):
+    """An `RSAPrivateKey` implementation based on a key managed by Key Vault.
+
+    :param str key_name: The name of the key vault key to use.
+    :param key_client: The client that will be used to communicate with Key Vault.
+    :type key_client: KeyClient
+    """
+
+    def __init__(self, key_name: str, key_client: "KeyClient") -> None:  # pylint:disable=unused-argument
```

Comment:
hmm wondering if client should come first.
Alternatively we make them both keyword only and avoid the ordering problem.

Reply:
I just updated the constructor a bit to take a crypto client, key ID, and key material, based on [.NET's implementation](https://github.com/Azure/azure-sdk-for-net/blob/8a110f564fb2a8982d4d7b7128978c80a22680bf/sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/RSAKeyVault.cs#L23). This is intended to be used via a [create_rsa_private_key](https://github.com/Azure/azure-sdk-for-python/blob/492eb3ad181dd071a0991e66629d9c969b5e7950/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py#L216) method on the `CryptographyClient`.

Reply:
After more thought, though, it doesn't look like we actually have any use for a key ID, since the crypto client will have fetched the key material anyway. I think I'll remove it unless we think there's a good reason to accept it